src/libsyntax_ext/format.rs

```diff
@@ -251,23 +253,49 @@ impl<'a, 'b> Context<'a, 'b> {
 
     fn describe_num_args(&self) -> String {
         match self.args.len() {
-            0 => "no arguments given".to_string(),
-            1 => "there is 1 argument".to_string(),
-            x => format!("there are {} arguments", x),
+            0 => "no arguments were given".to_string(),
+            1 => "there is only 1 argument".to_string(),
+            x => format!("there are only {} arguments", x),
         }
     }
 
+    /// Handle invalid references to positional arguments. Output different
+    /// errors for the case where all arguments are positional and for when
+    /// there are named arguments in the format string.
+    fn report_invalid_references(&self) {
+        let mut refs: Vec<String> = self.invalid_refs
+                                        .iter()
+                                        .map(|r| r.to_string())
+                                        .collect();
+
+        let msg = if self.names.is_empty() {
```

Comment:
Add another check for `self.args.iter().filter_map(|arg| /* Some(arg) if numbered positional argument */).count() == 0` here, so that if there are arguments like `{1}`, it goes to the else branch and it is explicit about the arguments that are wrong.

Reply:
Seems like the position is already resolved in the parser, should we change it so we still have that information here?

Reply:
If you prefer, that's also a reasonable change.